theMind.js

```diff
@@ -1,35 +1,48 @@
 // the story of the samurai who woke hungover and needed to find his lost katana
 // before his daimio found out and made him cut his guts with his little sword...
 
+function Labyrinth (numberOfLines, numberOfColumns, wallsPerRow, startY, startX, goalY, goalX) {
```

Comment:
Choose an option from numberOfLines/Columns, X/Y, distanceFromTop/Left and use it when necessary.